packages/core/src/runtime.ts

```diff
@@ -272,10 +274,14 @@ export function workflowEntrypoint(
               let workflowRun = await world.runs.get(runId);
 
               if (workflowRun.status === 'pending') {
-                workflowRun = await world.runs.update(runId, {
-                  // This sets the `startedAt` timestamp at the database level
-                  status: 'running',
+                // Transition run to 'running' via event (event-sourced architecture)
+                const result = await world.events.create(runId, {
+                  eventType: 'run_started',
                 });
+                // Use the run entity from the event response (no extra get call needed)
+                if (result.run) {
```

Comment:
`run` should exist in response and we should throw if `run` is not returned instead of silently eating the error